### PR TITLE
Akka design and latest plugins

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Put Akka design on the Akka Management docs.
Use the latest sbt-plugin versions.

## Changes

* Akka Paradox template
* more https
* Version warning support
* Canonical URLs
* reordered Paradox groups so the default will be Java
* extended copyright

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -226,15 +226,18 @@ lazy val `integration-test-local` = project
...
   .disablePlugins(BintrayPlugin)
   .settings(
     name := "Akka Management",
-    paradoxGroups := Map("Language" -> Seq("Scala", "Java")),
-    paradoxTheme := Some(builtinParadoxTheme("generic")),
-    paradox in Compile := (paradox in Compile).dependsOn(LocalRootProject / ScalaUnidoc / doc).value,
-    paradoxProperties ++= Map(
+    publish / skip := true,
+    whitesourceIgnore := true,
+    paradoxGroups := Map("Language" -> Seq("Java", "Scala")),
```
Comment (raboof) [inline build.sbt]: Do we really want Java first? OK by me, but it's not what we do for most other projects AFAIK

File: build.sbt [hunk 2]
```diff
@@ -226,15 +226,18 @@ lazy val `integration-test-local` = project
...
+  .enablePlugins(AkkaParadoxPlugin, NoPublish)
   .disablePlugins(BintrayPlugin)
   .settings(
     name := "Akka Management",
-    paradoxGroups := Map("Language" -> Seq("Scala", "Java")),
-    paradoxTheme := Some(builtinParadoxTheme("generic")),
-    paradox in Compile := (paradox in Compile).dependsOn(LocalRootProject / ScalaUnidoc / doc).value,
-    paradoxProperties ++= Map(
+    publish / skip := true,
+    whitesourceIgnore := true,
```
Comment (raboof) [inline build.sbt]: Shouldn't this already be part of `NoPublish` above?

File: project/Common.scala [hunk 3]
```diff
@@ -10,15 +10,16 @@ object Common extends AutoPlugin {
...
+    organizationHomepage := Some(url("https://www.lightbend.com/")),
     startYear := Some(2017),
-    homepage := Some(url("https://github.com/akka/akka-management")),
+    homepage := Some(url("https://akka.io/")),
     scmInfo := Some(ScmInfo(url("https://github.com/akka/akka-management"), "git@github.com:akka/akka-management.git")),
     developers += Developer("contributors", "Contributors", "https://gitter.im/akka/dev", url("https://github.com/akka/akka-management/graphs/contributors")),
 
     licenses := Seq(("Apache-2.0", url("https://www.apache.org/licenses/LICENSE-2.0"))),
-    headerLicense := Some(HeaderLicense.Custom("Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>")),
+    headerLicense := Some(HeaderLicense.Custom("Copyright (C) 2017-2019 Lightbend Inc. <https://www.lightbend.com>")),
```
Comment (raboof) [inline project/Common.scala]: :+1: , but travis suggests some other files also need updating

File: project/Common.scala [hunk 4]
```diff
@@ -10,15 +10,16 @@ object Common extends AutoPlugin {
 
   override lazy val projectSettings =
     Dependencies.Common ++ Seq(
-    organization := "com.lightbend.akka.management", // FIXME proposing this change, then we can have "com.lightbend.akka.management|discovery" etc; we'd do management.commercial for example later
+    organization := "com.lightbend.akka.management",
     organizationName := "Lightbend Inc.",
+    organizationHomepage := Some(url("https://www.lightbend.com/")),
     startYear := Some(2017),
-    homepage := Some(url("https://github.com/akka/akka-management")),
+    homepage := Some(url("https://akka.io/")),
```
Comment (raboof) [inline project/Common.scala]: or to the root of the akka-management docs? Either :+1: by me

File: build.sbt (hunk 2 above)
Comment (ennru) [reply build.sbt]: I'll drop `NoPublish`, all other modules used `publish / skip` already anyway.

File: project/Common.scala (hunk 4 above)
Comment (ennru) [reply project/Common.scala]: We point to `akka.io` in other Akka modules.

File: build.sbt (hunk 1 above)
Comment (ennru) [reply build.sbt]: This is part of the roll-out :-)

**Comment by ennru** [comment]
@raboof Kubernetes builds fail. Might that be due to too aggressive plugin upgrades?

**Comment by raboof** [comment]
Looks like https://github.com/sbt/sbt-native-packager/issues/1202, an incompatibility between sbt-native-packager and alpine. I'm not sure what our recommended docker base image is these days - alpine doesn't seem great because of this and uClibc, the redhat/openshift ones might not be publicly available.

**Comment by ennru** [comment]
Travis is much happier after the rollback too native packager 1.3.1.

**Comment by TimMoore** [comment]
Coincidentally, Red Hat released a freely reusable base image yesterday: https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image

**Pull request merged**